src/Hakyll/Web/Template/Internal.hs

```diff
@@ -1,13 +1,12 @@
 --------------------------------------------------------------------------------
 -- | Module containing the template data structure
-{-# LANGUAGE DeriveDataTypeable         #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 module Hakyll.Web.Template.Internal
-    ( Template (..)
```

Comment:
The move was necessary because otherwise @samgd were running into cyclical module issues.  But if you need it we can expose the constructor from `Hakyll.Web.Template` or do some moving around such that the constructor is still exposed.